bpf/tests/Makefile

```diff
@@ -49,3 +53,6 @@ clean:
 	rm -f $(wildcard *.ll)
 	rm -f $(wildcard *.o)
 	rm -f $(wildcard *.i)
+	rm -f $(wildcard *.d)
+
+-include $(TEST_OBJECTS:.o=.d)
```

Comment:
What is this include statement doing? What's the problem with the previous version of the definition?

Reply:
The `.d` files which is generated by clang contains a list of all files which were used during the previous compilation of the .o file including all header file (.h). They look something like this:

```
some-test.o: some-test.c lib/ip.h bpf/include/something.h
```

So this line then takes all files from `TEST_OBJECTS` and replaces the .o with a .d. And we then include all of these .d files as makefiles into this one.

The effect this has is that it makes `make` aware of all of the .h that the root .c includes without having to explicitly list all of them. So any time a .h file is edited calling `make` afterwards will trigger recompilation of all .c files which include it.

> What's the problem with the previous version of the definition?

Before this change, modifying the .h would result in make saying `make: Nothing to be done for 'all'`. You would have to call `make clean` to force recompilation or make a bogus change to the .c file so it would recognize something changed. Leading to everyone being confused why changes in .h files did not change the outcome of tests, until they realize this and then have to manually rebuild every time.

Reply:
I see, thanks for the explanation. Does the `include` line need to be prefixed with `-`? This ensures that nonzero exit statuses are ignored, right? But it's not clear to me that this is required, and I don't see it in the guide.

Reply:
Yes, the `-` ignores errors. The .d files are generated after the initial call to clang, so when compiling the first time no .d files exist yet. In the guide they use the `wildcard` function to avoid this issue, but this works as well.

Reply:
Hmm, when I try this locally I don't see any `*.d` files.

Reply:
In case it makes a difference:
```
$ clang --version
Ubuntu clang version 14.0.0-1ubuntu1.1
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
```

Reply:
^ PEBKAC, wasn't running make for the right dir.